app/models/ems_refresh/save_inventory_container.rb

```diff
@@ -150,25 +217,28 @@ def save_container_routes_inventory(ems, hashes, target = nil)
     store_ids_for_new_records(ems.container_routes, hashes, :ems_ref)
   end
 
-  def save_container_nodes_inventory(ems, hashes, target = nil)
-    return if hashes.nil?
-    target = ems if target.nil?
-
-    ems.container_nodes.reset
-    deletes = if target.kind_of?(ExtManagementSystem)
-                :use_association
-              else
-                []
-              end
-
-    save_inventory_multi(ems.container_nodes, hashes, deletes, [:ems_ref],
-                         [:labels, :tags, :computer_system, :container_conditions,
-                          :additional_attributes], [:namespace])
-    store_ids_for_new_records(ems.container_nodes, hashes, :ems_ref)
+  def graph_container_nodes_inventory(ems, hashes, target = nil)
+    hashes.to_a.each do |h|
+      h = h.except(:labels, :tags, :additional_attributes) # TODO children
+      h = h.except(:namespace)
+      node = @inv_collections[:container_nodes].lazy_find(h[:ems_ref])
+      graph_container_node_conditions_inventory(node, h.delete(:container_conditions))
+      graph_computer_system_inventory(node, h.delete(:computer_system))
+      @inv_collections[:container_nodes].build(h)
+    end
   end
 
-  def save_computer_system_inventory(container_node, hash, _target = nil)
-    save_inventory_single(:computer_system, container_node, hash, [:hardware, :operating_system])
+  def graph_computer_system_inventory(parent, hash, _target = nil)
+    return if hash.nil?
+    hash = hash.merge(:managed_entity => parent)
+    # TODO: there is probably is shorter way to link them?
```

Comment:
Thanks.  I don't really mind IC == table, it's actually a nice constraint that makes it easier to think about.
was just wondering if I missed something big.